### PR TITLE
EZP-30013: --env option not passed to parallel processes of `ezplatform:reindex` Command

### PR DESCRIPTION
**Pull request by mateuszbieniek**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30013](https://jira.ez.no/browse/EZP-30013)
| **Bug**| yes
| **New feature**    | no
| **Target version** | `6.7`, `6.13`, `7.4`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php [hunk 1]
```diff
@@ -149,6 +154,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $commit = !$input->getOption('no-commit');
         $iterationCount = $input->getOption('iteration-count');
         $this->siteaccess = $input->getOption('siteaccess');
+        $this->env = $this->getContainer()->getParameter('kernel.environment');
```
[the remark this answers is not in the document]
Comment (mateuszbieniek) [reply eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php]: Done https://github.com/ezsystems/ezpublish-kernel/pull/2526/commits/924c246fdf324e54a5cb2adc7600a03bc4017f06

**Comment by gggeek** [comment]
As a side note: when generating SF (sub)commands, I found out that it helps to also add the following snippet:
```
        if (!$kernel->isDebug()) {
            $subProcessArgs[] = '--no-debug';
        }
```
to propagate usage of the `--no-debug` flag

**Comment by andrerom** [comment]
> This starts to be a maintenance hell, we should consider some uniform handling of these.

@alongosz Fully agree,  there is a followup to do here, one option:
In Sf 4 there is something for this afaik, so maybe backport this feature to `symfony-tools` and take advantage of it already.

**Comment by mateuszbieniek** [comment]
@andrerom I'm on it

**Comment by lserwatka** [comment]
Could you merge it up?

**Comment by kmadejski** [comment]
@lserwatka merged up in https://github.com/ezsystems/ezpublish-kernel/commit/bbf43326e288d39b276cd9fa4864985d9940483b / https://github.com/ezsystems/ezpublish-kernel/commit/9f8c3471e388f54d26b75ef338bc3321c1eeabfd / https://github.com/ezsystems/ezpublish-kernel/commit/d6c2968bb2dd68b4cff8a1d2296317a2aca33822 / https://github.com/ezsystems/ezpublish-kernel/commit/4f484f6a864468502e579400db63b7a0c9ac1cb5.

**Comment by gggeek** [comment]
Sorry for the late comment, and maybe I am missing something evident here, but it seems wrong for this command to access directly the spi layer bypassing the Repo services, as it introduces a strong tie to the DB schema... Is it due to the fact that there is currently no way to force the Repo services to execute db queries even when the Search is configured to use a separate system such as Solr?

**Comment by mateuszbieniek** [comment]
@gggeek it seems that your comment is out of the scope of this PR.
Maybe you could create a public ticket for it here: jira.ez.no?

**Comment by gggeek** [comment]
@mateuszbieniek done: https://jira.ez.no/browse/EZP-30558

**Comment by mateuszbieniek** [comment]
@gggeek thanks!

**Pull request merged**